consensus/src/liveness/proposal_generator.rs

```diff
@@ -168,6 +175,9 @@ pub struct ProposalGenerator {
     // Last round that a proposal was generated
     last_round_generated: Round,
     quorum_store_enabled: bool,
+
+    sys_txn_providers: Vec<Arc<dyn SysTxnProvider>>,
```

Comment:
why this is a vector? each provider only returns a single txn? I thought we can just have a single provider that collects all system txns

Reply:
so
```
trait SysTxnReader {
    fn read(&self) -> Vec<Arc<SystemTransaction>>;
}
```
?

Reply:
maybe we should make it similar to the mempool pull api? something like pull(max_num, max_bytes, exclude)

Reply:
fixed